packages/Kokkos/test/tstPointCloud.cpp

```diff
@@ -0,0 +1,196 @@
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \file   tstPointCloud.cpp
+ * \author Stuart Slattery
+ * \brief  Point Cloud unit tests.
+ */
+//---------------------------------------------------------------------------//
+
+#include <DTK_PointCloud.hpp>
+
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_ScalarTraits.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+#include <limits>
+
+//---------------------------------------------------------------------------//
+// FUNCTOR HELPERS
+//---------------------------------------------------------------------------//
+//! Create global ids.
+template <class SC, class LO, class GO, class NO>
+class CreateIdsFunctor
+{
+  public:
+    using PointCloud = DataTransferKit::PointCloud<SC, LO, GO, NO>;
+    using DeviceType = typename NO::device_type;
+    using GlobalIdView = Kokkos::View<GO *, DeviceType>;
+
+    CreateIdsFunctor( GlobalIdView global_ids, int comm_rank )
+        : _global_ids( global_ids )
+        , _comm_rank( comm_rank )
+    { /* ... */
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()( const size_t p ) const
+    {
+        _global_ids( p ) = _comm_rank * _global_ids.extent( 0 ) + p;
+    }
+
+  private:
+    GlobalIdView _global_ids;
+    int _comm_rank;
+};
+
+//---------------------------------------------------------------------------//
+// TEST TEMPLATE DECLARATIONS
+//---------------------------------------------------------------------------//
+// Point cloud test.
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( PointCloud, basic, SC, LO, GO, NO )
+{
+    // Test types.
+    using PointCloud = DataTransferKit::PointCloud<SC, LO, GO, NO>;
+    using DeviceType = typename NO::device_type;
+    using GlobalIdView = Kokkos::View<GO *, DeviceType>;
+    using CoordinateView = Kokkos::View<SC **, DeviceType>;
+    using ExecutionSpace = typename PointCloud::execution_space;
+
+    // Get the communicator.
+    auto comm = Teuchos::DefaultComm<int>::getComm();
+    int comm_rank = comm->getRank();
+    int comm_size = comm->getSize();
+
+    // Create global ids.
+    size_t local_num_points = 10000;
+    GlobalIdView global_ids( "global_ids", local_num_points );
+    CreateIdsFunctor<SC, LO, GO, NO> id_functor( global_ids, comm_rank );
+    Kokkos::parallel_for(
+        "fill_gids", Kokkos::RangePolicy<ExecutionSpace>( 0, local_num_points ),
+        id_functor );
+    Kokkos::fence();
+
+    // Create point coordinates.
+    size_t space_dim = 3;
+    CoordinateView coordinates( "coordinates", local_num_points, space_dim );
+    Kokkos::Random_XorShift64_Pool<ExecutionSpace> rng_pool( 34998321 );
+    for ( size_t d = 0; d < 3; ++d )
+    {
+        auto sv = Kokkos::subview( coordinates, Kokkos::ALL(), d );
+        Kokkos::fill_random( sv, rng_pool, local_num_points );
+    }
+    Kokkos::fence();
```

Comment:
because the `fill_random` calls occur consecutively and on separate dimensions of the array so we don't have to worry about any ill-effects of threaded code between calls. If I was doing other stuff in this loop you would possibly have to call `fence()` every time.